pkg/cluster/docker_desktop.go

```diff
@@ -403,3 +403,57 @@ func (c DockerDesktopClient) settingsForWrite(settings interface{}) interface{}
 
 	return settings
 }
+
+type clientRequest struct {
+	client  HTTPClient
+	method  string
+	url     string
+	headers map[string]string
+	body    func() (*bytes.Buffer, error)
+}
+
+func status2xx(resp *http.Response) bool {
+	return resp.StatusCode >= 200 && resp.StatusCode <= 204
+}
+
+func (c DockerDesktopClient) tryRequests(label string, requests []clientRequest) (resp *http.Response, err error) {
```

Comment:
i think my main concern about this approach is how it handles errors. if both requests fail, how do we know which is the "right" error? presumably, the error message may contain important info.

Reply:
i wonder if we should bite the bullet and do a docker-desktop version check at the top...i feel like we now have multiple levels of "retry different versions of the route" and it's getting difficult to reason about which one is expected for which version. But I could be convinced otherwise...

Reply:
Yeah, that's a possibility. I'd need to go track down which versions they changed in, but that wouldn't be too difficult.